pkg/proxy/iptables/proxier.go

```diff
@@ -1099,6 +1126,52 @@ func (proxier *Proxier) syncProxyRules() {
 			activeNATChains[svcXlbChain] = true
 		}
 
+		if proxier.localDetector.IsImplemented() && svcChain != "" {
+			// We have a LocalDetector, so fill in the rules in the SVC chain to
+			// redirect external traffic (or simply masquerade it if we don't
+			// have a separate external chain).
+			var externalTrafficDest utiliptables.Chain
+			if svcXlbChain != "" {
```

Comment:
So I guess I never stated this explicitly, but one of the ideas in the rewrite is that `svcChain` gets set if and only if we are using a SVC chain and `svcXlbChain` gets set if and only if we are using an XLB chain. So then later on, anything that might add rules to the SVC chain checks `svcChain != ""` and anything that might add rules to the XLB chain checks `svcXlbChain != ""`.

So in this case in particular, `if svcInfo.NodeLocalExternal()` happens to be equivalent at this point in the PR, but later on, the rule for when we do and don't create an XLB chain changes, so `if svcInfo.NodeLocalExternal()` would no longer be correct, but `if svcXlbChain != ""` still is.

Maybe I should make explicit `bool`s rather than using the chain name variables as booleans.